pkg/auth/azure.go

```diff
@@ -278,34 +229,43 @@ func (a *Azure) getBearerTokenForApplication() (string, error) {
 	return tokenResponse.AccessToken, nil
 }
 
-func (a *Azure) getBearerTokenOnBehalfOfUser(token string) (string, error) {
-	form := url.Values{}
-	form.Add("client_id", a.clientID)
-	form.Add("client_secret", a.clientSecret)
-	form.Add("scope", "https://graph.microsoft.com/.default")
-	form.Add("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer")
-	form.Add("requested_token_use", "on_behalf_of")
-	form.Add("assertion", token)
+func (a *Azure) GetGroupID(groupMail string) (string, error){
+	token, err := a.getBearerTokenForApplication()
+	if err != nil {
+		return "",err
+	}
 
-	req, err := http.NewRequest(http.MethodPost, endpoints.AzureAD(a.tenantID).TokenURL, strings.NewReader(form.Encode()))
+	params := url.Values{}
+	params.Add("$select", "id,displayName,mail")
+	params.Add("$filter", fmt.Sprintf("mail eq '%v'", groupMail))
+
+	req, err := http.NewRequest(http.MethodGet, 
+		AzureGroupsEndpoint+ "?"+params.Encode(),
+		nil)
 	if err != nil {
+		fmt.Print(err)
 		return "", err
 	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %v", token))
 
 	httpClient := &http.Client{
 		Timeout: time.Second * 10,
 	}
 
 	response, err := httpClient.Do(req)
 	if err != nil {
+		fmt.Print(err)
 		return "", err
 	}
 
-	var tokenResponse TokenResponse
-	if err := json.NewDecoder(response.Body).Decode(&tokenResponse); err != nil {
+	var groupsResponse AzureGroupsWithIDResponse
+	if err := json.NewDecoder(response.Body).Decode(&groupsResponse); err != nil {
+		fmt.Print(err)
 		return "", err
 	}
-
-	logrus.Debugf("Successfully retrieved on-behalf-of token")
-	return tokenResponse.AccessToken, nil
-}
+	var groupID = ""
+	if(len(groupsResponse.Groups)>0){
```

Comment:
Burde vi ikke feile hvis vi ikke får noen grupper? For nå kan vel man i teorien bli admin ved å ha en tom gruppe 🤔

Reply:
Du er riktig, jeg har fikst det